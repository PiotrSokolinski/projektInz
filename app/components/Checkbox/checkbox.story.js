import React from 'react'

import { storiesOf } from '@storybook/react'

import Checkbox from '.'

storiesOf('Checkbox', module)
  .add('default', () => <Checkbox />)
  .add('default disabled', () => <Checkbox disabled />)
  .add('default checked', () => <Checkbox checked />)
  .add('default with label', () => <Checkbox label="Example label" />)
  .add('default with label checked', () => <Checkbox label="Example label" checked />)
  .add('rounded', () => <Checkbox rounded />)
  .add('rounded disabled', () => <Checkbox disabled rounded />)
  .add('rounded checked', () => <Checkbox checked rounded />)
  .add('rounded label', () => <Checkbox label="Example label" rounded />)
  .add('rounded label checked', () => <Checkbox label="Example label" checked rounded />)
