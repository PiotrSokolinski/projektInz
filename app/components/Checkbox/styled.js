import styled from 'styled-components'

import { Tick } from 'styled-icons/typicons/Tick'

import { Colors } from '../../themes/index'

const getCheckboxColor = props => {
  if (props.checked) return Colors.cerulean
  if (props.rounded) return Colors.edward
  if (props.isBackgroundGray) return Colors.concrete
  return Colors.white
}

export const TickIcon = styled(Tick)`
  color: ${Colors.white};
  margin-top: -3px;
`

export const Container = styled.label`
  color: ${Colors.black};
  cursor: ${props => (props.disabled ? 'auto' : 'pointer')};
  display: flex;
  font-size: 13px;
  letter-spacing: 0.02em;
  line-height: 19px;
  user-select: none;
  .checkmark {
    background: ${props => getCheckboxColor(props)};
  }
`

export const Input = styled.input`
  opacity: 0;
  height: 0;
  width: 0;
`

export const Checkmark = styled.span`
  border-radius: ${props => (props.rounded ? '100px' : '3px')};
  border: 1px solid ${props => (props.rounded ? 'transparent' : Colors.coral)};
  display: flex;
  height: 18px;
  justify-content: space-around;
  margin-right: 6px;
  width: 18px;

  ${props =>
    props.disabled &&
    !props &&
    `
    border-color: ${Colors.edward};
  `}
`

export const Label = styled.span`
  ${props => (props.checked ? 'font-weight: bold;' : '')};
  &::after {
    display: block;
    content: attr(label);
    font-weight: bold;
    height: 0;
    overflow: hidden;
    visibility: hidden;
  }
`
