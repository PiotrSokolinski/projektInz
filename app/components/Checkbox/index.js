/**
 *
 * Checkbox
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { Label, Container, Checkmark, Input, TickIcon } from './styled'

const Checkbox = ({
  label,
  checked,
  onChange,
  disabled,
  isControlled,
  rounded,
  isBackgroundGray,
  inputProps,
  ...otherProps
}) => (
  <Container
    checked={checked}
    disabled={disabled}
    isBackgroundGray={isBackgroundGray}
    onChange={!disabled ? onChange : null}
    rounded={rounded}
    {...otherProps}
  >
    <Checkmark className="checkmark" disabled={disabled} rounded={rounded}>
      {checked && <TickIcon size="22" />}
    </Checkmark>
    {!isControlled && <Input type="checkbox" {...inputProps} />}
    <Label className="checkbox-label" checked={checked} label={label}>
      {label}
    </Label>
  </Container>
)

Checkbox.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.number]),
  isBackgroundGray: PropTypes.bool,
  isControlled: PropTypes.bool,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  onChange: PropTypes.func,
  otherProps: PropTypes.object,
  rounded: PropTypes.bool,
  inputProps: PropTypes.object,
}

Checkbox.defaultProps = {
  checked: false,
  disabled: false,
  isBackgroundGray: false,
  isControlled: false,
  label: '',
  onChange: () => {},
  rounded: false,
}

export default Checkbox
