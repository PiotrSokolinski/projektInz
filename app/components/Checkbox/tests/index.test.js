/**
 *
 * Tests for Checkbox
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'

import Checkbox from '../index'

describe('<Checkbox />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(<Checkbox />)
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Checkbox />)
    expect(firstChild).toMatchSnapshot()
  })

  it('Should render with label, if it is passed to props', () => {
    const {
      container: { firstChild },
    } = render(<Checkbox label="Test label" />)
    expect(firstChild).toMatchSnapshot()
  })

  it('Should render without input, if it is controlled in parent', () => {
    const {
      container: { firstChild },
    } = render(<Checkbox label="Test label" isControlled />)
    expect(firstChild).toMatchSnapshot()
  })
  it('Should render with img, if it is checked', () => {
    const {
      container: { firstChild },
    } = render(<Checkbox label="Test label" checked />)
    expect(firstChild).toMatchSnapshot()
  })
})
