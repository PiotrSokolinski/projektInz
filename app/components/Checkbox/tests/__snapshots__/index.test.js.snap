// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Should render and match the snapshot 1`] = `
<label
  class="styled__Container-gubfwv-1 hLuRwd"
>
  <span
    class="checkmark styled__Checkmark-gubfwv-3 dCsIEv"
  />
  <input
    class="styled__Input-gubfwv-2 kVKIKx"
    type="checkbox"
  />
  <span
    class="checkbox-label styled__Label-gubfwv-4 JqeJh"
    label=""
  />
</label>
`;

exports[`<Checkbox /> Should render with img, if it is checked 1`] = `
<label
  class="styled__Container-gubfwv-1 jjBLUt"
>
  <span
    class="checkmark styled__Checkmark-gubfwv-3 dCsIEv"
  >
    <svg
      aria-hidden="true"
      class="styled__TickIcon-gubfwv-0 lhQSam StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      height="22"
      viewBox="0 0 24 24"
      width="22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16.972 6.251a1.999 1.999 0 00-2.72.777l-3.713 6.682-2.125-2.125a2 2 0 10-2.828 2.828l4 4c.378.379.888.587 1.414.587l.277-.02a2 2 0 001.471-1.009l5-9a2 2 0 00-.776-2.72z"
      />
    </svg>
  </span>
  <input
    class="styled__Input-gubfwv-2 kVKIKx"
    type="checkbox"
  />
  <span
    class="checkbox-label styled__Label-gubfwv-4 hhmCJw"
    label="Test label"
  >
    Test label
  </span>
</label>
`;

exports[`<Checkbox /> Should render with label, if it is passed to props 1`] = `
<label
  class="styled__Container-gubfwv-1 hLuRwd"
>
  <span
    class="checkmark styled__Checkmark-gubfwv-3 dCsIEv"
  />
  <input
    class="styled__Input-gubfwv-2 kVKIKx"
    type="checkbox"
  />
  <span
    class="checkbox-label styled__Label-gubfwv-4 JqeJh"
    label="Test label"
  >
    Test label
  </span>
</label>
`;

exports[`<Checkbox /> Should render without input, if it is controlled in parent 1`] = `
<label
  class="styled__Container-gubfwv-1 hLuRwd"
>
  <span
    class="checkmark styled__Checkmark-gubfwv-3 dCsIEv"
  />
  <span
    class="checkbox-label styled__Label-gubfwv-4 JqeJh"
    label="Test label"
  >
    Test label
  </span>
</label>
`;
